azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -1160,6 +1300,29 @@ func resourceLinuxVirtualMachineDelete(d *schema.ResourceData, meta interface{})
 		log.Printf("[DEBUG] Skipping Deleting OS Disk from Linux Virtual Machine %q (Resource Group %q)..", id.Name, id.ResourceGroup)
 	}
 
+	if features.VMDataDiskBeta() {
+		deleteDataDiskOnDeletion := meta.(*clients.Client).Features.VirtualMachine.DeleteDataDiskOnDeletion
+
+		// delete any disks created with the VM if feature toggled to do so. "existing" disks are not affected.
+		if deleteDataDiskOnDeletion {
+			if props := existing.VirtualMachineProperties; props != nil && props.StorageProfile != nil && props.StorageProfile.DataDisks != nil {
+				dataDisks := *props.StorageProfile.DataDisks
+				for _, v := range dataDisks {
+					if v.CreateOption == compute.DiskCreateOptionTypesEmpty && v.Name != nil {
+						deleteFuture, err := disksClient.Delete(ctx, id.ResourceGroup, *v.Name)
```

Comment:
we don't need to for this initial version, but it'd be worth making this easier to do this in parallel in the future - perhaps it's worth pulling these into a list, then calling a separate method to make this easier to parallelise in the future and also reusable?